### PR TITLE
enables updating status heart beats

### PR DESCRIPTION
**Pull request by xh3b4sd**
This PR updates the `lastHeartbeatTime` property of cluster status conditions on a regular basis. During testing the update threshold was set to 10 minutes. Doing the same for nodes and versions will be done later in other PRs. 

```
$ kubectl get kvmconfigs.provider.giantswarm.io zeub4 -o yaml | yq r - status
cluster:
  conditions:
  - lastHeartbeatTime: 2018-11-30T21:06:20.000715676Z
    lastTransitionTime: 2018-11-30T21:06:20.000715676Z
    status: "True"
    type: Creating
  network:
    cidr: ""
  nodes:
  - lastHeartbeatTime: 2018-11-30T21:10:02.650795653Z
    lastTransitionTime: 2018-11-30T21:10:02.650795793Z
    name: master-xaao4-7b4565b458-k6t9p
    version: 3.1.0
  - lastHeartbeatTime: 2018-11-30T21:10:02.650796838Z
    lastTransitionTime: 2018-11-30T21:10:02.650796915Z
    name: worker-dfub7-67ff7586cc-89tj5
    version: 3.1.0
  resources: null
  versions: []
```

```
$ kubectl get kvmconfigs.provider.giantswarm.io zeub4 -o yaml | yq r - status
cluster:
  conditions:
  - lastHeartbeatTime: 2018-11-30T21:20:17.032222463Z
    lastTransitionTime: 2018-11-30T21:06:20.000715676Z
    status: "True"
    type: Creating
  network:
    cidr: ""
  nodes:
  - lastHeartbeatTime: 2018-11-30T21:10:02.650795653Z
    lastTransitionTime: 2018-11-30T21:10:02.650795793Z
    name: master-xaao4-7b4565b458-k6t9p
    version: 3.1.0
  - lastHeartbeatTime: 2018-11-30T21:10:02.650796838Z
    lastTransitionTime: 2018-11-30T21:10:02.650796915Z
    name: worker-dfub7-67ff7586cc-89tj5
    version: 3.1.0
  resources: null
  versions: []
```

```
$ kubectl get kvmconfigs.provider.giantswarm.io zeub4 -o yaml | yq r - status
cluster:
  conditions:
  - lastHeartbeatTime: 2018-11-30T21:25:19.157118866Z
    lastTransitionTime: 2018-11-30T21:25:19.157118866Z
    status: "True"
    type: Created
  network:
    cidr: ""
  nodes:
  - lastHeartbeatTime: 2018-11-30T21:25:19.125299243Z
    lastTransitionTime: 2018-11-30T21:25:19.125299364Z
    name: master-xaao4-7b4565b458-k6t9p
    version: 3.1.0
  - lastHeartbeatTime: 2018-11-30T21:25:19.125300458Z
    lastTransitionTime: 2018-11-30T21:25:19.125300534Z
    name: worker-dfub7-67ff7586cc-89tj5
    version: 3.1.0
  - lastHeartbeatTime: 2018-11-30T21:25:19.125301088Z
    lastTransitionTime: 2018-11-30T21:25:19.125301166Z
    name: worker-k49hv-77555478cc-t6hmc
    version: 3.1.0
  - lastHeartbeatTime: 2018-11-30T21:25:19.125301736Z
    lastTransitionTime: 2018-11-30T21:25:19.125301809Z
    name: worker-mbv84-5cfbc6c9f-gdqhj
    version: 3.1.0
  resources: null
  versions:
  - date: 0001-01-01T00:00:00Z
    lastHeartbeatTime: 2018-11-30T21:25:19.401696487Z
    lastTransitionTime: 2018-11-30T21:25:19.401696572Z
    semver: 3.1.0
```

```
$ kubectl get kvmconfigs.provider.giantswarm.io zeub4 -o yaml | yq r - status
cluster:
  conditions:
  - lastHeartbeatTime: 2018-11-30T21:35:21.264914898Z
    lastTransitionTime: 2018-11-30T21:25:19.157118866Z
    status: "True"
    type: Created
  network:
    cidr: ""
  nodes:
  - lastHeartbeatTime: 2018-11-30T21:25:19.125299243Z
    lastTransitionTime: 2018-11-30T21:25:19.125299364Z
    name: master-xaao4-7b4565b458-k6t9p
    version: 3.1.0
  - lastHeartbeatTime: 2018-11-30T21:25:19.125300458Z
    lastTransitionTime: 2018-11-30T21:25:19.125300534Z
    name: worker-dfub7-67ff7586cc-89tj5
    version: 3.1.0
  - lastHeartbeatTime: 2018-11-30T21:25:19.125301088Z
    lastTransitionTime: 2018-11-30T21:25:19.125301166Z
    name: worker-k49hv-77555478cc-t6hmc
    version: 3.1.0
  - lastHeartbeatTime: 2018-11-30T21:25:19.125301736Z
    lastTransitionTime: 2018-11-30T21:25:19.125301809Z
    name: worker-mbv84-5cfbc6c9f-gdqhj
    version: 3.1.0
  resources: null
  versions:
  - date: 0001-01-01T00:00:00Z
    lastHeartbeatTime: 2018-11-30T21:25:19.401696487Z
    lastTransitionTime: 2018-11-30T21:25:19.401696572Z
    semver: 3.1.0
```

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -37,8 +37,8 @@ required = [
   branch = "master"
   name = "github.com/docker/distribution"
 
-[[constraint]]
-  branch = "master"
+[[override]]
+  branch = "extend-status-time"
   name = "github.com/giantswarm/apiextensions"
```
Comment (xh3b4sd) [inline Gopkg.toml]: See https://github.com/giantswarm/apiextensions/pull/177.

File: resource.go [hunk 2]
```diff
@@ -17,6 +17,10 @@ import (
 	"k8s.io/client-go/rest"
 )
 
+const (
+	HeartBeatUpdateThreshold = 30 * time.Minute
```
Comment (rossf7) [inline resource.go]: Not deep into this story so I wanted to check this. We're aiming for a heartbeat of 30 mins? The 10 mins mentioned in PR is just for testing.

Also for my interest are we starting with 30 mins and then reducing as we get experience? Or is this the "final" value.

**Review by rossf7 — APPROVED** [review]
Changes LGTM. Just a query on the update threshold.

File: resource.go (hunk 2 above)
Comment (xh3b4sd) [reply resource.go]: That is hard to decide. We get a new event with every change so updating the heartbeat all the time would deadlock itself. So we have to threshold it. That approach feels wrong or incomplete. Maybe the reconciliation has to be improved or maybe the mechanism. Not certain right now. This is how we can make it work now and when there are more strict requirements we have to get together and see what we can do.

Comment (rossf7) [reply resource.go]: Ack. Thanks for the explanation. Makes sense to iterate on the threshold as we get more experience with it.

**Pull request merged**